lib/couchbase3.js

```diff
@@ -140,3 +145,231 @@ Couchbase.prototype.disconnect = function(callback) {
   this.cluster = null;
   return promise.then(resolved, callback);
 };
+
+/**
+ * Create a new model instance for the given data
+ * @param {String} model The model name
+ * @param {Object} data The model data
+ * @param {Function} [callback] The callback function
+ */
+Couchbase.prototype.create = function (model, data, callback) {
+  var self = this;
+  debug("Couchbase.CREATE: " + JSON.stringify([model, data]));
+
+  if (callback == null) {
+    callback = noop;
+  }
+
+  // Handle with promise.
+  var promise;
+  var resolved = function(res) {
+    return callback(null, res);
+  };
+
+  // The cached promise.
+  promise = this._connection;
+  if (promise == null) {
+    debug('No connections. Try to connect...');
+    promise = self.connect(callback);
+  }
+
+  //bucket id
+  var id = self.getIdValue(model, data);
+
+  promise = promise.then(function(bucket) {
+    Promise.promisifyAll(bucket);
```

Comment:
Interesting. I doubt we want to use this but could you:

- Do this in `connect()`.
- Do a benchmark (on your local) and tell me which is faster, the callbacks or the "promisified" functions.